processing/src/test/java/io/druid/segment/IndexMergerTest.java

```diff
@@ -813,6 +817,46 @@ public void testDisjointDimMerge() throws Exception
     Assert.assertArrayEquals(new Object[]{1L}, boatList.get(4).getMetrics());
   }
 
+  @Test
+  public void testMismatchedMetrics() throws IOException
+  {
+    IncrementalIndex index1 = IncrementalIndexTest.createIndex(new AggregatorFactory[]{
```

Comment:
Updated to close all test indices and added additional cases.  
All of this is still part of one test merge, but still exhibits the trait where the specified set of metrics is a superset of the union of index metrics. I can break this into multiple test merges if you think this will be more clear.